plugins/rikki/heroeslounge/classes/heroes/HeroUpdater.php

```diff
@@ -276,168 +89,77 @@ public static function updateTalentsForHero($hero, $version = null)
-                    $file = fopen($talent_image_path.DS.$image_url, "w+");
-                    fputs($file, $talent_icon);
-                    fclose($file);
-                    $talent->image_url = $image_url;
-                    $talent->save();
-                    Resizer::open($talent_image_path.DS.$image_url)
-                        ->resize(32, 32)
-                        ->save($talent_image_path.DS.$image_url, 100);
-                } else {
-                    Log::error('Failed to get image for talent '.$talent_name);
-                }
+    public static function addTranslationsToHeroes()
+    {
+        $hero_list = Self::getHeroesList();
+        foreach ($hero_list as $hero_entry) {
+            $hero = Hero::where('title', $hero_entry['name'])->first();
+            if ($hero) {
+                $hero->translations = implode(",", $hero_entry['translations']);
+                $hero->save();
             }
         }
     }
 
     public static function getHeroesList()
     {
-        sleep(1);
-    	$ch = curl_init("https://api.masterleague.net/heroes/?format=json");
-    	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-    	$hero_list_json = curl_exec($ch);
-    	curl_close($ch);
-    	$decoded = json_decode($hero_list_json, true);
-    	$hero_list = $decoded['results'];
-    	while (array_key_exists('next', $decoded) && $decoded['next'] != null) {
-    		sleep(1);
-    		$ch = curl_init($decoded['next']);
-    		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-    		$hero_list_json = curl_exec($ch);
-    		curl_close($ch);
-    		$decoded = json_decode($hero_list_json, true);
-    		$hero_list = array_merge($hero_list, $decoded['results']);
-    	}
-    	return $hero_list;
-    }
+        $ch = curl_init("https://api.heroesprofile.com/openApi/Heroes");
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        $hero_list_json = curl_exec($ch);
+        curl_close($ch);
+        $hero_list = json_decode($hero_list_json, true);
 
-    public static function addMasterleagueIds()
-    {
-    	$hero_list = HeroUpdater::getHeroesList();
-    	foreach ($hero_list as $hero_entry) {
-    		$hero = Hero::where('title', $hero_entry['name'])->first();
-    		if ($hero) {
-    			$hero->masterleague_id = $hero_entry['id'];
-    		} else {
-    			$hero = Hero::where('translations', 'LIKE', '%'.$hero_entry['name'].'%')->first();
-    			if ($hero) {
-    				$hero->masterleague_id = $hero_entry['id'];
-    			} else {
-    				trigger_error($hero_entry['name']);
-    			}
-    		}
-    		$hero->save();
-    	}
+        return $hero_list;
     }
 
-    public static function addTranslationsToHeroes()
+    public static function getHero($hero_name)
     {
-        set_time_limit(600);
+        // There are naming exceptions for TLV and Cho.
+        if ($hero_name == 'thelostvikings') {
+            $hero_name = 'lostvikings';
+        } else if ($hero_name == 'cho') {
+            $hero_name = 'chogall';
+        }
```

Comment:
This can be refactored into a function HeroNameForHeropatchnotes() or something similar.